suite-native/notifications/src/components/TransactionNotification.tsx

```diff
@@ -0,0 +1,141 @@
+} from '@suite-native/navigation';
+import { Icon } from '@trezor/icons';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { useNativeStyles, prepareNativeStyle } from '@trezor/styles/src';
+
+import { AnnouncementNotification } from './AnnouncementNotification';
+
+type TransactionNotificationProps = {
+    txid: string;
+    accountKey: AccountKey;
+    isHiddenAutomatically?: boolean;
+};
+
+type TransactionNotificationDescriptionProps = {
+    amount: string | number;
+    transactionType: TransactionType;
+    networkSymbol: NetworkSymbol;
+    targetAddress?: string;
+};
+
+type TransactionTypeProperties = {
+    title?: string;
+    preposition?: string;
+};
+
+const addressContainerStyle = prepareNativeStyle(_ => ({
+    maxWidth: '35%',
+}));
+
+const transactionTypeToContentMap = {
+    recv: { title: 'Incoming transaction', preposition: 'from' },
+    sent: { title: 'Sending transaction', preposition: 'to' },
+    self: { title: 'Sending transaction', preposition: 'to' },
+    joint: { title: 'Sending transaction', preposition: 'to' },
+    failed: { title: 'Sending transaction', preposition: 'to' },
+    unknown: { title: 'Sending transaction', preposition: 'to' },
+} as const satisfies Record<TransactionType, TransactionTypeProperties>;
+
+export const TransactionNotificationDescription = ({
+    amount,
+    targetAddress,
+    transactionType,
+    networkSymbol,
+}: TransactionNotificationDescriptionProps) => {
+    const { applyStyle } = useNativeStyles();
+    const { preposition } = transactionTypeToContentMap[transactionType];
+    return (
+        <Box flexDirection="row">
+            <CryptoAmountFormatter
+                value={amount}
+                network={networkSymbol}
+                isBalance={false}
+                color="gray500"
+                variant="label"
+            />
+            <Text color="gray500" variant="label">
+                {` ${preposition} `}
+            </Text>
+            {targetAddress && (
+                <Box style={applyStyle(addressContainerStyle)}>
+                    <AccountAddressFormatter
+                        value={targetAddress}
+                        variant="label"
+                        color="gray500"
+                    />
+                </Box>
+            )}
+        </Box>
+    );
+};
+
+export const TransactionNotification = ({
+    txid,
+    accountKey,
+    isHiddenAutomatically = true,
+}: TransactionNotificationProps) => {
+    const navigation =
+        useNavigation<StackNavigationProps<RootStackParamList, RootStackRoutes.AppTabs>>();
+
+    const transaction = useSelector((state: TransactionsRootState) =>
+        selectTransactionByTxidAndAccountKey(state, txid, accountKey),
+    );
+
+    const transactionTargetAddress = useSelector((state: TransactionsRootState) =>
+        selectTransactionFirstTargetAddress(state, txid, accountKey),
+    );
+
+    if (!transaction) return null;
+
+    const { title } =
```

Comment:
Shouldn't return null if type is not in content map?

Reply:
I would prefer to fallback to uknown, null is problematic because it's very hard to debug if user report something strange.

Reply:
Yes, or use the Error message from atoms, that might be best

Reply:
There is no way that title will be undefined. The object is correctly typed. If there would be any of the transaction types missing, it would shout. example:

https://user-images.githubusercontent.com/26143964/220620046-16817efa-979c-43f7-8e45-79a81654e5cb.mov

Reply:
Yes, so you can remove the fallback then 😉

Since there is need to have a fallback, either use unknown or show error. I know that currently all types are supported, but it won't hurt to be on the safe side. I guess the fallback is never going to be used anyway if I use your argument, we should remove it then 😅

If you just fallback to unknown, it would make sense, but it doesn't make sense to fallback to `recv`, that's just wrong.

Reply:
Yeah. Since it's typed correctly, it doesn't need the fallback case. If you remove it, typescript doesn't scream

```suggestion
    const { title } = transactionTypeToContentMap[transaction.type];
```

Reply:
fixed [0dfad91](https://github.com/trezor/trezor-suite/commit/0dfad91c31f18abfb3a328936724ab34c064a83d)